### PR TITLE
#20 Automated Test #2

### LINKED ISSUE #20: Test#2
User can duplicate a comment (comment is created with all data from parent's comment and unique number)

### PR DESCRIPTION
**Pull request by oteplova**
Test #2: 
User can duplicate a comment (comment is created with all data from parent's comment and unique number)

Closes #20

### CODE REVIEW

**Comment by oteplova** [comment]
@extsoft please check.

**Comment by oteplova** [comment]
@extsoft  I've mistakenly push changes from branch #22 to branch #20 and I don't know how to revert them. What I should do in such cases?

**Comment by oteplova** [comment]
@extsoft  I've reverted changes for Test #3

File: src/main/java/net/comments/objects/CommentTable.java [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import java.util.List;
 
 public interface CommentTable {
-    void selectComment(int commentsNumber);
+    void selectComment(int commentNumber);
+
+    String getTextSelectedComment(int commentNumber);
```
Comment (extsoft) [inline src/main/java/net/comments/objects/CommentTable.java]: Please rename methods as I can't understand what it does.

File: src/main/java/net/comments/objects/BCommentManagerDuplicate.java [hunk 2]
```diff
@@ -1,6 +1,15 @@
 package net.comments.objects;
 
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
 public class BCommentManagerDuplicate implements CommentManager {
```
Comment (extsoft) [inline src/main/java/net/comments/objects/BCommentManagerDuplicate.java]: Why do you use a separate class for implementing each interface methods? Please implement 3 methods in the one class.

File: src/main/java/net/comments/objects/BCommentManagerDuplicate.java [hunk 3]
```diff
@@ -13,4 +22,10 @@ public void saveAndReturn(String number, String text, String state, String categ
         throw new UnsupportedOperationException("Please implement");
     }
 
+    public void saveAnReturn(String newCommentNumber) {
+        this.driver.findElement(By.xpath("//*[@id = \"Number\"]")).clear();
+        this.driver.findElement(By.xpath("//*[@id = \"Number\"]")).sendKeys(newCommentNumber);
```
Comment (extsoft) [inline src/main/java/net/comments/objects/BCommentManagerDuplicate.java]: You need use separate class to work with updating comments data. `Save & Return` is only a button to save changes.

**Review by extsoft — CHANGES_REQUESTED** [review]
@oteplova Please fix my comments.

**Comment by extsoft** [comment]
> @extsoft I've mistakenly push changes from branch #22 to branch #20 and I don't know how to revert them. What I should do in such cases?

@oteplova is it still relevant?

**Comment by oteplova** [comment]
> @oteplova is it still relevant?

No, I've already resolved this issue.

**Comment by oteplova** [comment]
@extsoft I've fixed comments. Please check.

**Review by extsoft — APPROVED** [review]
@oteplova great job!

**Pull request merged**